### PR TITLE
Pacifies all pets spawned by Xenobiology's Stabilized Gold

### LINKED ISSUE #11328: [Bug?]: Stabilized Gold Immortal Pet can attack
### BYOND Version

_No response_

### Round Date

2024/13/08

### Round ID

49634

### Test merges

_No response_

### Describe the bug

I'm unsure if this is a bug, a feature or unintended behaviour but basically, if you select the snake as your companion on the Stabilized Gold you pretty much got a immortal animal that can ventcrawl, give poison, make you bleed and brute damage, quite unstoppable killing machine since it can just keep returning to attack, specially if the wearer is well positionated, saw them single-handely kill three officers and a lot of doctors due to their instant respawn.

### What did you expect?

I expected for the companions to be harmless, i was unaware they were actually able to attack since i've only seen normal pets.

### Reproducible steps

1. Get stabilized gold extract
2. Get sentience potion?(unless the extract already does that)
3. Become an inmortal snake that can poison, bleed and attack ad infinitum

### PR DESCRIPTION
**Pull request by aramix273**
## About The Pull Request
for all I can tell, the pet spawned by stabilized gold should be a pacifist
the pseudo-immortality is biggest argument in this sense
it's still a good question why there are hostile animals in friendly paths and why they have the FRIENDLY_SPAWN tag despite their attacks (perhaps referring to their default AI rather than capacity to harm, which is kind of an edge case considering sentience)

## Why It's Good For The Game

Fixes: #11328 

you can no longer have stabilized gold pets that can bite and poison you then come back each time you kill them
though they can still ventcrawl, pull and strip
took this route rather than removing the snake and lizard from the pool outright because:
- it allows them to be pets still,
- this should cover any other cases if pets were to be added to this

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/user-attachments/assets/d21432e3-595f-4e47-93a8-e3c9f3d5dcf3

</details>

## Changelog
:cl: Aramix
tweak: pacified all pets spawned by Xenobiology's Stabilized Gold
/:cl:

### CODE REVIEW

**Comment by aramix273** [comment]
should ideally solve #11328 too (testing evidence shows so) while keeping code/mechanic changes local to stabilized gold and without modifying ``simple_animal`` code

**Comment by EvilDragonfiend** [comment]
why don't you just give them pacifist trait?

**Comment by aramix273** [comment]
> why don't you just give them pacifist trait?

did not come to mind to use it
apparently the parrot does not care about the proc that I registered that is supposed to cancel attacks
so, pacifism would help, albeit only for attacking mobs
apparently, even though the parrot has 0 ``obj_damage``, it can still hurt walls and other structures while ``melee_damage`` is non-zero,, so that has to be set to 0 too

**Comment by aramix273** [comment]
applied my last observations and had to handle a particular case of parrots being able to just change their ``melee_damage`` via "Toggle Mode"
this is getting a bit convoluted for the purpose of also disabling such a pet's capacity to damage structures, so if the PR's scope or implementation is getting a bit snowflakey, it could be reduced to pacification against just mobs, at expense of potential later abuse to break structures such as consoles or whatnot

File: code/modules/research/xenobiology/crossbreeding/_status_effects.dm [hunk 1]
```diff
@@ -938,6 +938,10 @@
 	var/obj/item/slimecross/stabilized/gold/linked = linked_extract
 	if(QDELETED(familiar))
 		familiar = new linked.mob_type(get_turf(owner.loc))
+		familiar.a_intent = "help"
```
Comment (EvilDragonfiend) [inline code/modules/research/xenobiology/crossbreeding/_status_effects.dm]: INTENT_HELP

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]

**Review by EvilDragonfiend — APPROVED** [review]

**Comment by RDS88-beeman** [comment]
As (I would like to imagine) the primary abuser of this bullshit, please hit burning yellow next.

**Pull request merged**